docs/topics/class-methods.rst

```diff
@@ -0,0 +1,29 @@
+===========================
+Class Instantiation Methods
+===========================
+
+These methods create an instance of their implementer class by 
+extracting the components needed for it from the argument that the method takes.
+
+.. py:classmethod:: from_crawler(cls, crawler)
+
+    Factory method that if present, is used to create an instance of the implementer class
+    using a :class:`~scrapy.crawler.Crawler`. It must return a new instance
+    of the implementer class. The Crawler object is needed in order to provide 
+    access to all Scrapy core components like settings and signals; It is a 
+    way for the implenter class to access them and hook its functionality into Scrapy.
+
+    :param crawler: crawler that uses this middleware
+    :type crawler: :class:`~scrapy.crawler.Crawler` object
+
+
+.. py:classmethod:: from_settings(cls, settings)
+
+    This class method is used by Scrapy to create an instance of the implementer class
+    using the settings passed as arguments.
+    In most implementations of this method, not all settings given will be used 
+    to instantiate the class, but only the ones needed in the specific scenario.
```

Comment:
I don’t think this bit is needed. It is up to the user which settings they want to use, and using a lot or 1 should make no difference.

On the other hand, in this definition I miss the fact that this class method will not be called at all if `from_crawler` is also defined, which I believe it a rather important detail.